### PR TITLE
Log configuration at start up

### PR DESCRIPTION
**Pull request by mem**
When the program is starting up, log the configuration parameters. This helps with debugging and support.

Be careful not to log the API secret.

This is a no-op change, but it's a little noisy becuase of the introduction of the config struct.

### CODE REVIEW

File: cmd/synthetic-monitoring-agent/secret.go [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+// String returns a redacted version of the secret.
+func (s Secret) String() string {
+	return "<redacted>"
+}
+
+// Set sets the value of the secret.
+func (s *Secret) Set(value string) error {
+	*s = Secret(value)
+	return nil
+}
```
Comment (roobre) [inline cmd/synthetic-monitoring-agent/secret.go:15]: Does `Set(string) error` make this implement any interface?

**Review by roobre — APPROVED** [review]
Having a `string` implement `Stringer` is a pretty clever way of redacting!

File: cmd/synthetic-monitoring-agent/secret.go (hunk 1 above)
Comment (mem) [reply cmd/synthetic-monitoring-agent/secret.go:15]: ah, yeah, it's what flag.Value wants. It _also_ wants String() for that matter.

Comment (mem) [reply cmd/synthetic-monitoring-agent/secret.go:15]: It's needed for line 83 to work.

**Pull request merged**